src/test/java/cloud/eppo/EppoClientBanditTest.java

```diff
@@ -0,0 +1,336 @@
+package cloud.eppo;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import cloud.eppo.helpers.*;
+import cloud.eppo.logging.Assignment;
+import cloud.eppo.logging.AssignmentLogger;
+import cloud.eppo.logging.BanditAssignment;
+import cloud.eppo.logging.BanditLogger;
+import cloud.eppo.ufc.dto.*;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Stream;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.MockedStatic;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EppoClientBanditTest {
+  private static final Logger log = LoggerFactory.getLogger(EppoClientBanditTest.class);
+  private static final String DUMMY_BANDIT_API_KEY =
+      "dummy-bandits-api-key"; // Will load bandit-flags-v1
+  private static final String TEST_HOST =
+      "https://us-central1-eppo-qa.cloudfunctions.net/serveGitHubRacTestFile";
+  private static final ObjectMapper mapper = new ObjectMapper().registerModule(module());
+
+  private static final AssignmentLogger mockAssignmentLogger = mock(AssignmentLogger.class);
+  private static final BanditLogger mockBanditLogger = mock(BanditLogger.class);
+  private static final Date testStart = new Date();
+
+  // TODO: possibly consolidate code between this and the non-bandit test
```

Comment:
Thinking of leaving this as a TODO for now until everything is said and done. I don't want to prematurely combine parts of tests only to have things diverge once non-bandit-only (for now) things like caching is layered on.